lib/ash/seed.ex

```diff
@@ -143,6 +143,134 @@ defmodule Ash.Seed do
     end
   end
 
+  @doc """
+  Performs an upsert operation on the data layer of a resource with the provided input and identities.
+  The usage is the same as `seed!/1`, but it will update the record if it already exists.
+
+  ```elixir
+  Ash.Seed.upsert!(%User{email: 'test@gmail.com', name: 'Test'}, [:email])
+  ```
+  """
+  def upsert!(%{__meta__: %{state: :loaded}} = input, _identities) do
+    input
+  end
+
+  def upsert!(%resource{} = input, identities) do
```

Comment:
What should be put here is an identity name, or nil (meaning upsert via primary key). So like `upsert!(_, identity \\ nil)`

Reply:
Then we look up the identity fields prior to upsert.